dandi/upload.py

```diff
@@ -453,6 +453,19 @@ def _bids_discover_and_validate(
             )
             if valid:
                 validated_datasets.append(bd)
+        invalid_datasets = []
+        for i in bids_datasets_to_validate:
+            if i not in validated_datasets:
+                invalid_datasets.append(str(i))
+        if invalid_datasets:
+            raise RuntimeError(
```

Comment:
should not raise exception if validation = 'ignore' (as exception suggests).

and if `ignore` it is worthwhile to lgr.warning (instead of exception) that information.  So just format the message, then then 

```python
if validation == 'ignore':
    lgr.warning(msg)
else:
    raise RuntimeError(msg) 
```

Reply:
not really, the logic is encoded here:

```python
valid = is_valid(
                validator_result,
                allow_missing_files=validation == "ignore",
                allow_invalid_filenames=validation == "ignore",
            )
```

The function starts off with this:

```python
if allow_invalid_filenames and allow_missing_files:
        return True
```

so there's no case for which behaviour would not be equivalent.

Reply:
I don't think any of that is related to my concern. Am I correct that having this exception here, it would be raised if any found BIDS dataset is invalid, correct?  If yes -- I will not be able to upload **any** file from any dataset since dandi cli would just crash before due to this exception, right? and that is what I am trying to avoid

Reply:
Right, but if `validation == "ignore"` all datasets will always be evaluated as valid. The only case where you could get an exception is if the validation code breaks and one is raised, to bypass running validation altogether we have `validation == "skip"`.

Reply:
ok, as too me it is too much of assumptions on the code behavior somewhere not nearby. but ok, let's just proceed for now and see how it goes.

Meanwhile please address typing issues which are still there!